test/integration_test.py

```diff
@@ -283,11 +284,20 @@ def test_snapshot_listing(self):
         """
         Test with a small page size to be sure paging works
         """
+        snapshots = self._public_tdr_client.snapshot_names_by_id()
         page_size = 5
+        if len(snapshots) == 1:
+            log.warning('Public service account only has access to 1 snapshot')
+            page_size = 1
+        elif len(snapshots) < page_size:
```

Comment:
I think this should be 
```suggestion
        elif len(snapshots) <= page_size:
```
Otherwise the assertion on line 300 will fail when `len(shapshots) == page_size`